source/collimation.f90

```diff
@@ -1799,18 +1801,16 @@ subroutine collimate_start_sample(nsample)
     if(firstrun) write(RHIClosses_unit,'(a)') '# 1=name 2=turn 3=s 4=x 5=xp 6=y 7=yp 8=dp/p 9=type'
   end if
 
-!++  Copy new particles to tracking arrays. Also add the orbit offset at
-!++  start of ring!
-
+  ! Copy new particles to tracking arrays. Also add the orbit offset at start of ring!
   do i = 1, napx00
-    xv(1,i)  = c1e3 *  myx(i+(j-1)*napx00) + torbx(1)              !hr08
-    yv(1,i)  = c1e3 * myxp(i+(j-1)*napx00) + torbxp(1)             !hr08
-    xv(2,i)  = c1e3 *  myy(i+(j-1)*napx00) + torby(1)              !hr08
-    yv(2,i)  = c1e3 * myyp(i+(j-1)*napx00) + torbyp(1)             !hr08
-
-!JULY2005 assignation of the proper bunch length
-    sigmv(i) = mys(i+(j-1)*napx00)
-    ejv(i)   = myp(i+(j-1)*napx00)
+    if(do_thisdis /= 0) then
+      xv(1,i)  = c1e3 *  myx(i+(j-1)*napx00) + torbx(1)
+      yv(1,i)  = c1e3 * myxp(i+(j-1)*napx00) + torbxp(1)
+      xv(2,i)  = c1e3 *  myy(i+(j-1)*napx00) + torby(1)
+      yv(2,i)  = c1e3 * myyp(i+(j-1)*napx00) + torbyp(1)
+      sigmv(i) = mys(i+(j-1)*napx00)
+      ejv(i)   = myp(i+(j-1)*napx00)
+    end if
```

Comment:
I would maybe swap the order of the loop/if-statement:
```
if(do_thisdis /= 0) then
   do i=1,napx00
      [...]
      ejv(j) = [...]
   end do
end if

do i=1,napx00
```

Reply:
This is exactly what triggered the gfortran compiler bug in the closed orbit addition. This is therefore risky since we have no proper tests for this.

Reply:
I manually inlined the calculation, but I have mo way of checking it.